src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_NuclearReactor.java

```diff
@@ -342,155 +356,121 @@ public boolean turnCasingActive(final boolean status) {
 		return true;
 	}
 
+	public FluidStack[] getStoredFluidsAsArray() {	
+		final ArrayList<FluidStack> tFluids = this.getStoredFluids();
+		FluidStack[] aStored = new FluidStack[tFluids.size()];	
+		for (int i = 0; i < aStored.length; i++) {
+			aStored[i] = tFluids.get(i);
+		}
+		return aStored;
+	}
+
+	public int getStoredFuel(GT_Recipe aRecipe) {
+		int aFuelStored = 0;
+		FluidStack aFuelFluid = null;
+		for (FluidStack aFluidInput : aRecipe.mFluidInputs) {
+			if (!aFluidInput.getFluid().equals(NUCLIDE.Li2BeF4.getFluid())) {
+				aFuelFluid = aFluidInput;
+				break;
+			}
+		}
+		if (aFuelFluid != null) {
+			for (GT_MetaTileEntity_Hatch_Input aInputHatch : this.mInputHatches) {
+				if (aInputHatch.getFluid() != null && aInputHatch.getFluidAmount() > 0) {
+					if (aInputHatch.getFluid().isFluidEqual(aFuelFluid)) {
+						aFuelStored += aInputHatch.getFluidAmount();
+					}
+				}
+			}
+		}		
+		return aFuelStored;
+	}
+
 	@Override
 	public boolean checkRecipe(final ItemStack aStack) {
-		final ArrayList<FluidStack> tFluids = this.getStoredFluids();
-		final Collection<GT_Recipe> tRecipeList = GTPP_Recipe.GTPP_Recipe_Map.sLiquidFluorineThoriumReactorRecipes.mRecipeList;
-		if((tFluids.size() > 0) && (tRecipeList != null)) { //Does input hatch have a LFTR fuel?
+		// Warm up for 4~ minutes
+		Logger.WARNING("Checking LFTR recipes.");
+		if (mEfficiency < this.getMaxEfficiency(null)) {
+			this.mProgresstime = 0;
+			this.mMaxProgresstime = 1;
+			this.mEfficiencyIncrease = 2;
+			Logger.WARNING("Warming Up! "+this.mEfficiency+"/"+this.getMaxEfficiency(null));
+			return true;
+		}
+		Logger.WARNING("Warmed up, checking LFTR recipes.");
+
+		final FluidStack[] tFluids = getStoredFluidsAsArray();
+		final Collection<GT_Recipe> tRecipeList = getRecipeMap().mRecipeList;
+		if(tFluids.length > 0 && tRecipeList != null && tRecipeList.size() > 0) { //Does input hatch have a LFTR fuel?
 			Logger.WARNING("Found more than one input fluid and a list of valid recipes.");
+			boolean foundLi2bef4 = false;
+			// Find a valid recipe
+			GT_Recipe aFuelProcessing = this.findRecipe(getBaseMetaTileEntity(), mLastRecipe, true, 0, tFluids, new ItemStack[] {});
+			if (aFuelProcessing == null) {
+				Logger.WARNING("Did not find valid recipe for given inputs.");
+				return false;
+			}
+			else {
+				Logger.WARNING("Found recipe? "+(aFuelProcessing != null ? "true" : "false"));
+				for (FluidStack aFluidInput : aFuelProcessing.mFluidInputs) {
+					Logger.WARNING("Using "+aFluidInput.getLocalizedName());				
+				}
+			}
+			// Find li2bef4, Helium & Fluorine
 			for (final FluidStack hatchFluid1 : tFluids) { //Loops through hatches
```

Comment:
We probably don't need to check for existence of Li2BeF4. It's always part of these recipes so findRecipe has done it already.

Reply:
Fixed in next push.